Mathlib/SetTheory/Cardinal/Cofinality.lean

```diff
@@ -1215,3 +1215,19 @@ lemma sup_sequence_lt_omega1 {α} [Countable α] (o : α → Ordinal) (ho : ∀
 end Ordinal
 
 end Omega1
+
+namespace Ordinal
+
+theorem isLimit_of_not_isRegular_aleph {o : Ordinal} (h : ¬((aleph o).IsRegular)) :
+    o.IsLimit := by
+  have ho : o ≠ 0 := fun h0 ↦ False.elim <| (aleph_zero ▸ h0 ▸ h) isRegular_aleph0
+  contrapose! h
+  obtain ⟨p, rfl⟩ := ((zero_or_succ_or_limit o).resolve_left ho).resolve_right h
+  exact isRegular_aleph_succ p
+
+theorem ord_cof_pos_of_isLimit {o : Ordinal} (h : IsLimit o) : 0 < o.cof.ord :=
```

Comment:
I don't think this lemma makes much sense, when really, any non-zero ordinal has positive cofinality.

Reply:
Oh of course. What do you think of `ord_cof_pos : 0 < o.cof.ord ↔ 0 < o`? Generally `0 < o` is the better conclusion and `≠ 0` the assumption, so I'm not sure what the best formulation here is.

Reply:
I also don't think we need that. Isn't it just a composition of `c.cof = 0` iff `c = 0` and `c.ord = 0` iff `c = 0`?